src/main/scala/tech/cryptonomic/conseil/tezos/PlatformDiscoveryOperations.scala

```diff
@@ -0,0 +1,130 @@
+
+import scala.collection.JavaConverters._
+import scala.concurrent.{ExecutionContext, Future}
+
+
+object PlatformDiscoveryOperations {
+
+  lazy val dbHandle: PostgresProfile.backend.Database = DatabaseUtil.db
+
+  private val tables = List(Tables.Blocks, Tables.Accounts, Tables.OperationGroups, Tables.Operations, Tables.Fees)
+
+  def getNetworks(config: Config): List[Network] = {
+    config.getObject("platforms").asScala.flatMap {
+      case (platform, strippedConf) =>
+        strippedConf.atKey(platform).getObject(platform).asScala.map {
+          case (network, _) =>
+            Network(network, network.capitalize, platform, network)
+        }.toList
+    }.toList
+  }
+
+  def getEntities(network: String)(implicit ec: ExecutionContext): Future[List[Entity]] = {
+    ApiOperations.countAll.map { counts =>
+      tables.map(_.baseTableRow.tableName).flatMap { tableName =>
+        counts.get(tableName).map { tableCount =>
+          Entity(
+            name = tableName,
+            displayName = makeDisplayName(tableName),
+            count = tableCount,
+            network = network
+          )
+        }.toList
+      }
+    }
+  }
+
+  private def makeDisplayName(name: String): String = {
+    name.capitalize.replace("_", " ")
+  }
+
+  def tableAttributes(tableName: String)(implicit ec: ExecutionContext): Future[List[Attributes]] = {
+    val tablesMap = tables.map(table => table.baseTableRow.tableName -> table)
+
+    val res = tablesMap.find(_._1 == tableName).map {
+      case (_, table) =>
+        dbHandle.run {
+          val overallCount = TezosDb.count(table)
+          DBIO.sequence(
+            table.baseTableRow.create_*.map { col =>
+              for {
+                overallCnt <- overallCount
+                distinctCnt <-  TezosDb.countDistinct(table.baseTableRow.tableName, col.name)
+              } yield  (col, distinctCnt, overallCnt)
+            }
+          )
+        }.map { distinctCountList =>
+          distinctCountList.map {
+            case (col, distinctCount, overallCount) =>
+              Attributes(
+                name = col.name,
+                displayName = makeDisplayName(col.name),
+                dataType = mapType(col.tpe.toString),
+                cardinality = distinctCount,
+                keyType = if(distinctCount == overallCount) KeyType.UniqueKey else KeyType.NonKey,
+                entity = table.baseTableRow.tableName
+              )
+          }
+        }
+    }
+    futOpt(res).map(_.toList.flatten)
+  }
+
+  private def mapType(tpe: String): DataType = {
+    tpe match {
+      case "java.sql.Timestamp'" => DataType.DateTime
+      case "String'" => DataType.String
+      case "Int'" => DataType.Int
+      case "Long'" => DataType.LargeInt
+      case "Float'" | "Double'" | "scala.math.BigDecimal'" => DataType.Decimal
+      case "Boolean" => DataType.Boolean
+      case x if x.startsWith("Option[") => mapType(x.drop(7).dropRight(1))
```

Comment:
ouch... how rude 😏

Reply:
this part is tricky, as in the ticket defined there was no definition of nullable or optional field I've decided that I need to extract it somehow. Probably there is some regex which could help me but I don't know it yet and still looking for it 😄

Reply:
ok, found a regex and replaced this case 😄 also is worth to mention that I've only matched the cases we currently have in the DB(or are really easy to implement) so there is a possibility that this method will need to be extended in the future